client/src/router.js

```diff
@@ -1,144 +1,33 @@
 import { createWebHistory, createRouter } from 'vue-router';
 import { useAuthStore } from './stores/auth';
-
-import ChangePassword from './pages/ChangePassword.vue';
-import DataSourceStaticView from './pages/DataSourceStaticView.vue';
-import LogIn from './pages/LogIn.vue';
-import QuickSearchPage from './pages/QuickSearchPage.vue';
-import ResetPassword from './pages/ResetPassword.vue';
-import SearchResultPage from './pages/SearchResultPage.vue';
-import NotFound from './pages/NotFound.vue';
-
-import acronym from 'pdap-design-system/images/acronym.svg';
-
-const routes = [
-	{
-		path: '/',
-		component: QuickSearchPage,
-		name: 'QuickSearchPage',
-		// Use meta property on route to override meta tag defaults
-		// meta: { title: 'Police Data Accessibility Project - Search', metaTags: [{ property: 'og:title', title: 'Police Data Accessibility Project - Search' }] },
-	},
-	{
-		path: '/search',
-		component: SearchResultPage,
-		name: 'SearchResultPage',
-	},
-	{
-		path: '/search/:searchTerm/:location',
-		component: SearchResultPage,
-		name: 'SearchResultPage',
-	},
-	{
-		path: '/data-sources/:id',
-		component: DataSourceStaticView,
-		name: 'DataSourceStaticView',
-	},
-	{
-		path: '/login',
-		component: LogIn,
-		name: 'LogIn',
-	},
-	{
-		path: '/change-password',
-		component: ChangePassword,
-		name: 'ChangePassword',
-	},
-	{
-		path: '/reset-password/:token?',
-		component: ResetPassword,
-		name: 'ResetPassword',
-	},
-	{ path: '/:pathMatch(.*)*', name: 'not-found', component: NotFound },
-];
+import { routes, handleHotUpdate } from 'vue-router/auto-routes';
+import { refreshMetaTagsByRoute } from '@/util/routes.js';
 
 const router = createRouter({
 	history: createWebHistory(),
 	routes,
+	scrollBehavior(_to, _from, savedPosition) {
+		if (savedPosition) return savedPosition;
+		return { top: 0 };
+	},
```

Comment:
As with before, I think this would benefit from a comment -- it's not clear to me what `scrollBehavior` does -- is it saving where on a web page someone scrolled to so that it returns to there if the page refreshes?

Reply:
This is a built-in callback exposed on the VueRouter config object class. It's already documented [here](https://router.vuejs.org/guide/advanced/scroll-behavior.html), and in a JS-Doc comment that is revealed on hover in any modern IDE. So I think an extra comment here would be superfluous.

<img width="734" alt="Screenshot 2024-08-02 at 8 21 53 AM" src="https://github.com/user-attachments/assets/e9d1bc12-8218-48b6-9d98-588efeb4b709">

Reply:
Ah, I see what you mean. In this case, I didn't realize that this was from a third party application, rather than a custom function, and so my recommendation indeed tends more towards being superfluous.